private/eseses/psite.py

```diff
@@ -13,23 +13,36 @@
             print(out.read())
         sys.exit()
 
-url = 'http://geoapp03.ucsd.edu/gpseDB/psite?op=getNeuConversionSiteList'
+url = 'http://geoapp01.ucsd.edu:8080/gpseDB/psite?op=getNeuConversionSiteList'
 try:
     reader = csv.reader(urllib.urlopen(url), delimiter=' ')
-except Exception as e:
+except Exception as e:   # On error, print the saved version and exit
     if os.path.exists('out.json'):
         with open('out.json', 'r') as out:
             print(out.read())
     sys.exit()
 
+# save a copy of the web service response
+csvLines = []
+for line in reader:
+  csvLines.append(line)
+
+if len(csvLines) < 2:   # an empty response from web service
+     if os.path.exists('out.json'):   # print the saved copy
+        with open('out.json', 'r') as out:
+            print(out.read())
+        sys.exit() 
+
+# Got some csv from the url, now in cvsLines
 with open('out.csv', 'wb') as csvfile:
     wtr = csv.writer( csvfile )
     wtr.writerow( ('site', 'x', 'y' ))
     try:
-        for row in reader:
+        for row in csvLines:
             if "'Error'" in str(row): # use the old version if there is an error
                 with open('out.json', 'r') as out:
                     print(out.read())
+                    # should we sys.exit() at this point?  Why repeat for every row?  Why write csv if Error?
```

Comment:
This returns the content of the old out.json if there is an error reading new data.

Reply:
Yeah, but it's in the for row in csvLines loop, so I'm unclear if it encouters 'Error' why it carries on with the rest of the rows, writing them to the csv file -- shouldn't it sys.exit() after printing the old json, like it does above if there was an exception on the urlopen?  And maybe it's counting on only one occurrence of 'Error' but it looks like it would print the old json multiple times if there were multiple rows with 'Error.'

Reply:
Quick ping on this if I may.  Am I making any sense here?

Reply:
Ah, I missed this comment earlier. Yes, you're right, there isn't a need to continue after printing the old json output. Adding sys.exit() there makes sense.